### ISSUE TITLE
Get manual ready for 3.5.0

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags-manual#81]
[Milestone for 3.5.0](https://github.com/adventuregamestudio/ags-manual/milestone/1)

I am grouping in this issue all other issues that needs to be fixed so that we can use this manual along with the 3.5.0 release. The purpose of this issue is to discuss what more is missing to be added or if something needs to be removed. Specific details off each topic can be discussed on their own issues.

- [x] (#73) Have a proper entry for new Dictionary script type | [Dictionary.md](https://github.com/adventuregamestudio/ags-manual/wiki/Dictionary)
- [x] (#74) Have a proper entry for new Set script type | [Set.md](https://github.com/adventuregamestudio/ags-manual/wiki/Set)
- [x] (#75) Have a proper entry for new Camera functions | [Camera.md](https://github.com/adventuregamestudio/ags-manual/wiki/Camera)
- [x] (#76) Have a proper entry for new Viewport functions | [Viewport.md](https://github.com/adventuregamestudio/ags-manual/wiki/Viewport)
- [x] (#79) Write a new entry for Screen type | [Screen.md](https://github.com/adventuregamestudio/ags-manual/wiki/Screen)
- [x] (#77) Write an entry on the new Room Editor | [EditorRoom.md](https://github.com/adventuregamestudio/ags-manual/wiki/EditorRoom)
- [x] (#80) Fix missing AtScreenXY and AtRoomXY on existing items
- [x] (#82) GUI has new attributes accessible through script | [GUI.md](https://github.com/adventuregamestudio/ags-manual/wiki/GUI)
- [x] (#94) Adds missing entry for TextWindowGUI | [TextWindowGUI.md](https://github.com/adventuregamestudio/ags-manual/wiki/TextWindowGUI)
- [x] (#87) Add missing new Game script entries | [Game.md](https://github.com/adventuregamestudio/ags-manual/wiki/Game)
- [x] (#78) Write an entry on how to update an AGS game from previous version | [UpgradeTo350.md](https://github.com/adventuregamestudio/ags-manual/wiki/UpgradeTo350)
- [x] (#92) Intro/contents page does not exist in the CHM contents
- ~~[ ] (#91) Explain Script API and language switches | [Scripting.md](https://github.com/adventuregamestudio/ags-manual/wiki/Scripting)~~
- ~~[ ] (#102) Context search not working for non-static functions~~

**Comment by ivan-mogilko** [comment]
Hey, I think you may also use labels and/or milestones to group issues for a certain goal.
This way there is no need to add new entries in the list in one ticket. And easier to search for too.

**Comment by ericoporto** [comment]
I added a link to the milestone and grouped things inside of it. :) For now I will keep the list here, so if you see anything missing, open an issue and I will update here later.

**Comment by ericoporto** [comment]
I marked the ones I think have the needed content ready, but I need someone to review and confirm things are correct.

**Comment by ivan-mogilko** [comment]
Note that you may refer to Changes.txt in the main repository to find out which ScriptAPI was added.

**Comment by morganwillcock** [comment]
Need to re-add room properties somewhere (as displayed in the Editor) because the previous list was incomplete/incorrect and only listed on a tutorial page.

https://github.com/adventuregamestudio/ags-manual/wiki/acintro4/_compare/2d94b9d%5E...2d94b9d

**Comment by ivan-mogilko** [comment]
The closest, I think, is this page: https://github.com/adventuregamestudio/ags-manual/wiki/Settingupthegame

it lists things to set up for various entities, and General Settings too.

**Comment by Scummbuddy** [comment]
Hello everyone. You may not remember me (Scummbuddy, an old mod of the Beginners Tech forum), as I haven't been around since about 2010. I'm looking to get back into AGS and I have been running through the tutorial and happened to notice that it hasn't been updated since 3.0. I'd be more than happy to run through it with my experience and also my freshness to AGS. At least the tutorial could use a refresh (as mentioned by this issue) and I'd love to help out. There is a lot of good content already there that just needs some updating.

**Comment by morganwillcock** [comment]
@Scummbuddy it would be great it you could make some improvements to the tutorial. At the moment the manual releases are versioned separately to the Editor releases and it also builds to the website at https://adventuregamestudio.github.io/ags-manual/, so we should me able to make any changes available fairly quickly.

**Comment by ericoporto** [comment]
@Scummbuddy thanks for your contributions! Can you look into this guidelines (sorry for not yet adding it somewhere) for screenshots when changing the images? https://github.com/adventuregamestudio/ags-manual/issues/29#issuecomment-456600101

**Comment by Scummbuddy** [comment]
I sure will. I just looked at that briefly and I understand the concern about large image where there is little change. I've been working to keep all of the images the same size for consistency but when it was something small of detail being changed, I drew the red circles. I stopped doing updates for the moment as I go and cook breakfast but I will go back through those guidelines before I continue.

**Comment by Scummbuddy** [comment]
Hello again. I've read through it and I mostly agree with all the points. The introductory tutorial being mostly what I am working on, I feel that it is important to show the whole editor, along with the red circle call-outs when necessary, for the entire tutorial. As the user gets into the Advanced Tutorials and general manual, I feel at that point it could be a more proper time to cut down on the sizing of images as users should know where exactly the part of the editor being discussed is.

Was there anything in particular you wanted to stress to me? For instance, I see your comment on "no personally identifiable information" and I was actually about to replace the Merchant's name during the dialog screen as "Chris Jones" instead of "Derek" as a bit of fun. Should I not?

I could also go back and edit all of the images to not include the AGS Editor Window Header information I suppose.

**Comment by ericoporto** [comment]
By personally identifiable information I was thinking more about making sure there's not like a window with your government issued ID showing in a desktop capture.

I am all in for showing the full editor, but it's good to resize the window - don't use it maximized on a 2560x1440 monitor, try to resize to the minimal size where it's elements are clear and there's less wasted space. The bigger the image it's harder to read in small monitors, it's harder to resize the manual for a split view with the Editor (a common usage), and it's harder to read the screen on the phone.

These are more recommendations than rules. I welcome the contribution and am grateful for it, just trying to explain.

**Comment by Scummbuddy** [comment]
Thanks for all of that. :)

I certainly agree on not using Maximized windows in screenshots.

About the name, I was just trying to give a nice nod to the AGS-Puma-Man himself, Chris Jones. I reverted it back to 'Derek'.

I did make some additional adjustments to what I had already done, specifically making Script pictures simply be text, so that we're not breaking "Do not rely on images to provide information or context." which is important.

I've updated the 9 Introductory tutorials to 3.5, refreshed the 2 Scripting Tutorials, and linked the "Setting Up Your Game" and "Advanced Room Features" tutorials together through a common footer. I certainly hope my alterations doesn't affect the "push" to updating the AGS Editor manual.
